helix-rest/src/test/java/org/apache/helix/rest/client/TestCustomRestClient.java

```diff
@@ -161,4 +179,35 @@ protected JsonNode getJsonObject(HttpResponse httpResponse) throws IOException {
       return new ObjectMapper().readTree(_jsonResponse);
     }
   }
+
+  /**
+   * This MockCustomRestClientFactory is necessary to have for testing because once an INSTANCE is
+   * initialized in CustomRestClientFactory while running "mvn test" , it will no re-initialize the
+   * INSTANCE with new HelixProperty. Hence this class makes sure that new CustomRestClient will be
+   * created with the timeout set to the new value.
+   */
+  private static class MockCustomRestClientFactory extends CustomRestClientFactory {
+    private static final int DEFAULT_HTTP_REQUEST_TIMEOUT = 60 * 1000;
```

Comment:
Use long

Reply:
Please see my answer above. If you strongly feel we should do long, I can do it. But I think this way the code will be cleaner.

Reply:
in that case, using an int is fine. But can we avoid creating yet another constant here? I believe we already define it elsewhere.

Reply:
I added comments about it. I removed the default constant and directly uses the value in getSystemPropertyAsInt to avoid creating new constants. I found some other default values in for http request for MSDS related code. However, I am not sure if that is a good idea to have one default for everything. Anyway, please have a look and see if this makes sense.